src/ui/SimpleFilter/SimpleFilter.ts

```diff
@@ -0,0 +1,259 @@
+    if (this.getSelectedValues().length == 1) {
+      $$(title).text(l(this.getSelectedValues()[0]));
+    } else {
+      $$(title).text(this.options.title);
+    }
+    this.queryController.executeQuery();
+  }
+
+  private createCheckboxes(value: string) {
+    const checkbox = new Checkbox(() => {
+      this.handleCheckboxToggle();
+    }, l(this.getValueCaption(value)));
+    return { checkbox, value };
+  }
+
+  public getValueCaption(value: string): string {
+    let ret = value;
+
+    if (Utils.exists(this.options.valueCaption)) {
+      if (typeof this.options.valueCaption == 'object') {
+        ret = this.options.valueCaption[ret] || ret;
+      }
+    }
+    return ret;
+  }
+
+  public getSelected(): { checkbox: Checkbox, value: string }[] {
+    return _.filter(this.checkboxes, (object: { checkbox: Checkbox, value: string }) => object.checkbox.isSelected());
+  }
+
+  public getSelectedValues(): string[] {
+    return _.map(this.getSelected(), (object: { checkbox: Checkbox, value: string }) => object.value);
+  }
+
+  private buildCircleElement(): Dom {
+    this.circleElement = $$('span', { className: 'coveo-circle' }, this.getSelectedValues().length.toString());
+    return this.circleElement;
+  }
+
+  private createWrapper() {
+    if ($$(document.body).find('.coveo-filter-header-wrapper') == null) {
+      const wrapper = $$('div', { className: 'coveo-filter-header-wrapper' });
+      wrapper.insertBefore(this.element);
+      return wrapper;
+    } else {
+      const wrapper = $$(document.body).find('.coveo-filter-header-wrapper');
+      return $$(wrapper);
+    }
+  }
+
+  private handleRemoveFromBreadcrumb(label: string) {
+    _.each(this.checkboxes, (object: { checkbox: Checkbox, value: string }) => {
+      if (object.value == label) {
+        object.checkbox.toggle();
+      }
+    });
+  }
+
+  // ///////////////////////////////////////BREADCRUMB//////////////////////////////////////////////
+
+  private handlePopulateBreadcrumb(args: IPopulateBreadcrumbEventArgs) {
+
+
+    let elem = document.createElement('div');
+    $$(elem).addClass('coveo-simplefilter-breadcrumb');
+
+    let title = document.createElement('span');
+    $$(title).addClass('coveo-simplefilter-breadcrumb-title');
+    $$(title).text(this.options.title);
+    elem.appendChild(title);
+
+    let values = document.createElement('span');
+    $$(values).addClass('coveo-simplefilter-breadcrumb-values');
+    elem.appendChild(values);
+
+    _.each(this.getSelectedValues(), (test) => {
+      let value = document.createElement('span');
+      $$(value).addClass('coveo-simplefilter-breadcrumb-value');
+      $$(value).text(l(test));
+      values.appendChild(value);
+      let clear = document.createElement('span');
+      $$(clear).addClass('coveo-simplefilter-breadcrumb-clear');
+      value.appendChild(clear);
+      $$(value).on('click', () => this.handleRemoveFromBreadcrumb(value.textContent));
```

Comment:
Do you also handle if the user clicks on the global `Clear` button for all breadcrumb items ?